File: iRHSCBook/RHSCUser.m
//
//  RHSCUser.m
//  iRHSCBook
//
//  Created by Bruce Hunter on 2014-06-30.
//  Copyright (c) 2014 Richmond Hill Squash Club. All rights reserved.
//

#import "RHSCUser.h"

@interface RHSCUser ()

@property (nonatomic, strong) NSString *userid;
@property (nonatomic, strong) NSString *password;


@end

@implementation RHSCUser

BOOL loggedOn = NO;

-(id)initFromServer:(RHSCServer *)srvr userid:(NSString *)uid password:(NSString *)pwd
{
    self = [super init];
    self.userid = uid;
    self.password = pwd;
    _data = [self userFromJSON:srvr userid:self.userid password:self.password];
    return self;
}

- (RHSCMember *)userFromJSON:(RHSCServer *)server userid:(NSString *)uid password:(NSString *)pwd
{
    // Create a NSURLRequest with the given URL
    NSString *logonURL = [NSString stringWithFormat:@"Reserve/IOSUserLogonJSON.php?uid=%@&pwd=%@",uid,pwd];
    NSURL *target = [[NSURL alloc] initWithString:logonURL relativeToURL:server];
    NSURLRequest *request = [NSURLRequest requestWithURL:[target absoluteURL]
                                             cachePolicy:NSURLRequestReloadIgnoringLocalAndRemoteCacheData
                                         timeoutInterval:30.0];
	
    // Get the data
    NSError *error;
    NSURLResponse *response;
	NSData *data = [NSURLConnection sendSynchronousRequest:request returningResponse:&response error:&error];
    
    if (error == nil) {
        // Now create a NSDictionary from the JSON data
        NSDictionary *jsonDictionary = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
        
        // Get an array of dictionaries with the key "locations"
        NSArray *array = [jsonDictionary objectForKey:@"user"];
        // Iterate through the array of dictionaries
        for(NSDictionary *dict in array) {
            // Create a new Location object for each one and initialise it with information in the dictionary
            loggedOn = YES;
            return [[RHSCMember alloc] initWithJSONDictionary:dict];
        }
    }
    return nil;
}

-(BOOL)isLoggedOn
{
    return loggedOn;
}

@end

File: iRHSCBook/RHSCUser.swift
//
//  RHSCUser.swift
//  iRHSCBook
//
//  Created by Bruce Hunter on 2015-10-25.
//  Copyright © 2015 Bruce Hunter. All rights reserved.
//

import Foundation

@objc class RHSCUser : RHSCMember {
    
    static var loggedOn : Bool = false
    var userid : String? = nil
    var password : String? = nil

    init( forUserid uid : String, forPassword pwd : String) {
        userid = uid;
        password = pwd;
        super.init(name: userid!)
    }
    
    func validate(fromServer server:RHSCServer) -> Bool {
        
//        print(server.absoluteString)
        let url = URL(string: String.init(format: "Reserve20/IOSUserLogonJSON.php?uid=%@&pwd=%@", arguments: [userid!, password!]), relativeTo: server as URL )
//        print(url!.absoluteString)
        let sessionCfg = URLSession.shared.configuration
//        sessionCfg.timeoutIntervalForResource = 30.0
        let session = URLSession(configuration: sessionCfg)
        let semaphore_logon = DispatchSemaphore(value: 0)
        let task = session.dataTask(with: url!, completionHandler: { (data, response, error) -> Void in
            if error != nil {
                print("Error: \(error!.localizedDescription)")
            } else if data != nil {
//                print(NSString(data: data!, encoding: NSUTF8StringEncoding))
                do {
                    if let jsonDictionary = try JSONSerialization.jsonObject(with: data!, options: []) as? [String : Any] {
                        if let array = jsonDictionary["user"]! as? Array<[String : Any]> {
                            for dict in array {
                                RHSCUser.loggedOn = true
                                // Create a new Location object for each one and initialise it with information in the dictionary
                                
                                self.assign(fromJSONDictionary: dict)
                            }
                        } else {
                            print("user not found")
                        }
                    } else {
                        print("response was not JSON")
                    }
                } catch {
                    print(error)
                }
            }
            semaphore_logon.signal()
        })
        task.resume()
        _ = semaphore_logon.wait(timeout: DispatchTime.distantFuture)
        return RHSCUser.loggedOn
    }
    
    func isLoggedOn() -> Bool {
        return RHSCUser.loggedOn;
    }
    

    
}
